client/app/scripts/superdesk-authoring/authoring.js

```diff
@@ -894,7 +894,12 @@
                 $scope.close = function() {
                     _closing = true;
                     authoring.close($scope.item, $scope.origItem, $scope.dirty).then(function() {
-                        $location.url($scope.referrerUrl);
+                        if (sessionStorage.getItem('previewUrl')) {
+                            $location.url(sessionStorage.getItem('previewUrl'));
+                            sessionStorage.removeItem('previewUrl');
```

Comment:
OK, it's fine then. I also like the idea of moving it before the URL change line to avoid any doubt.